### DISCUSSION TITLE
Vaultwarden Web "My vault" is empty

### DISCUSSION THREAD

**Discussion by EyyupA** [dani-garcia/vaultwarden#5077] [Q&A] [answered]
### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.32.1
* Web-vault version: v2024.6.2c
* OS/Arch: linux/x86_64
* Running within a container: true (Base: Debian)
* Environment settings overridden: false
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Browser/Server Time Check: true
* Server/NTP Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.46.0
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** 

```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": true,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "_max_note_size": 10000,
  "_smtp_img_src": "cid:",
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_session_lifetime": 20,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "auth_request_purge_schedule": "30 * * * * *",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "***************",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://********************",
  "domain_origin": "*****://********************",
  "domain_path": "",
  "domain_set": true,
  "duo_context_purge_schedule": "30 * * * * *",
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "duo_use_iframe": false,
  "email_2fa_auto_fallback": false,
  "email_2fa_enforce_on_verified_invite": false,
  "email_attempts_limit": 3,
  "email_change_allowed": true,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 3 * * * *",
  "emergency_request_timeout_schedule": "0 7 * * * *",
  "enable_db_wal": true,
  "enable_websocket": true,
  "enforce_single_org_with_reset_pw_policy": false,
  "event_cleanup_schedule": "0 10 0 * * *",
  "events_days_retain": null,
  "experimental_client_feature_flags": "fido2-vault-credentials",
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "http_request_block_non_global_ips": true,
  "http_request_block_regex": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "increase_note_size_limit": false,
  "invitation_expiration_hours": 120,
  "invitation_org_name": "Allrounder Devs",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": "/data/vaultwarden.log",
  "log_level": "info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "",
  "org_events_enabled": false,
  "org_groups_enabled": false,
  "password_hints_allowed": true,
  "password_iterations": 100000,
  "push_enabled": true,
  "push_identity_uri": "https://identity.bitwarden.com",
  "push_installation_id": "***",
  "push_installation_key": "***",
  "push_relay_uri": "https://push.bitwarden.com",
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sendmail_command": null,
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": false,
  "signups_domains_whitelist": "",
  "signups_verify": true,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_embed_images": true,
  "smtp_explicit_tls": null,
  "smtp_from": "**********************",
  "smtp_from_name": "Vaultwarden",
  "smtp_host": "****************",
  "smtp_password": "***",
  "smtp_port": 587,
  "smtp_security": "starttls",
  "smtp_ssl": null,
  "smtp_timeout": 15,
  "smtp_username": "************************",
  "templates_folder": "data/templates",
  "tmp_folder": "data/tmp",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_sendmail": false,
  "use_syslog": false,
  "user_attachment_limit": null,
  "user_send_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "yubico_client_id": "80499",
  "yubico_secret_key": "***",
  "yubico_server": null
}
```
</details>

### Vaultwarden Build Version

v1.32.1

### Deployment method

Official Container Image

### Custom deployment method

version: '3.9'

services:
  vaultwarden:
    container_name: vaultwarden
    image: 'vaultwarden/server:1.32.1'
    labels:
      - "traefik.enable=true"
      # Main app routing
      - "traefik.http.routers.vaultwarden.rule=Host(`*********`)"
      - "traefik.http.services.vaultwarden.loadbalancer.server.port=80"
    environment:
      - WEBSOCKET_ENABLED=true
      - PUSH_ENABLED=true
      - PUSH_INSTALLATION_ID=************
      - PUSH_INSTALLATION_KEY=************
      - LOG_FILE=/data/vaultwarden.log
    volumes:
      - '/vw-data/:/data/'
    networks:
      traefik_web:

networks:
  traefik_web:
    external: true

### Reverse Proxy

traefik v3.2

### Host/Server Operating System

Linux

### Operating System Version

Debian

### Clients

Web Vault

### Client Version

2024.6.2c

### Steps To Reproduce

1. Go to 'Web Vault'
2. Login
3. No items from “My vault” are displayed there. Only those from the other collections

### Expected Result

All Items from all Collection.

### Actual Result

No items from “My vault” are displayed there. Only those from the other collections

### Screenshots or Videos

![2024-10-13 at 00 48 39](https://github.com/user-attachments/assets/bd5ebc31-bfad-4034-81a5-d203c8d42be7)
Here is a screenshot of Web Vault is display only those from the other collections.

**Comment by BlackDex** [comment]
What do you see in other clients. 
What happens if you use a different browser?

It probably is something with cache from either the reverse proxy or your browser. And it's might also be some browser extension which is causing strange issues.

**Reply by EyyupA** [reply]
The same problem with another browser. I used incognito windows. I have also emptied the Cloudflare cache.

**Reply by BlackDex** [reply]
And what about other clients?

**Reply by EyyupA** [reply]
On iOS or MacOs it works without problems. There will be displayed all items. This problem is only on web vault. My Web vault is also loading very long after login.

**Reply by BlackDex** [reply]
And what do you mean with MacOs? Which client?

Have you recently added or updated an item via any client besides the web-vault? What if you save those items again via the clients which do work and check the web-vault again?

**Reply by EyyupA** [reply]
![2024-10-13 at 12 18 43](https://github.com/user-attachments/assets/3fc081ee-9704-4a3f-9589-63fb03da99ec)
**Bitwarden:**
Version 2024.9.0
Shell 32.0.1
Renderer 128.0.6613.36
Node 20.16.0
Architecture arm64


The problem has existed for some time. I have only now been able to deal with it properly. 

In the meantime, I've had a look at the error messages I get. In the console I get: 
`[Error] Refused to execute a script because its hash, its nonce, or 'unsafe-inline' does not appear in the script-src directive of the Content Security Policy. (**************, line 0)`
`[Error] Refused to load blob:https://**************/7daebff1-dcb9-43f1-b64d-60c0c878bf20 because it does not appear in the script-src directive of the Content Security Policy.`
`[Error] Refused to execute a script because its hash, its nonce, or 'unsafe-inline' does not appear in the script-src directive of the Content Security Policy. (**************, line 0)`
`[Error] Refused to load blob:https://**************/1745b165-0a94-4eb4-bd59-42b66a929a6b because it does not appear in the script-src directive of the Content Security Policy.`
`[Warning] AES-CBC and AES-CTR do not provide authentication by default, and implementing it manually can result in minor, but serious mistakes. We recommended using authenticated encryption like AES-GCM to protect against chosen-ciphertext attacks. (main.7c1287a291f3bb48f2e0.js, line 11)`


I have checked whether I can assign this UUID. But unfortunately not. I exported my Vault and all organizations and searched for the UUID there. Unfortunately without success. 

Then I opened a new VW account and imported the export I made with Client into the new account. 
I noticed that not all passwords were imported. See:
![2024-10-13 at 12 23 19](https://github.com/user-attachments/assets/9ce0089a-cd4c-4177-a2ac-c5d5ca17f18f)

I have also tried to search for this UUID in the client, unfortunately without success. 
How can I find this UUID entry?

**Reply by BlackDex** [reply]
I think your reverse proxy is modifying the headers which set some security features of the browsers.
Vaultwarden already sends the correct headers to ensure the correct workings of Vaultwarden via the web-vault and other clients which might need specific headers or sometimes even omitting some headers.

Check your reverse proxy to be sure it doesn't override the headers Vaultwarden sends.

That `uuid` is not something i see during the loading of my environment. It might be a addon/extension which tries to do something which isn't allowed.

Here is an example of what Vaultwarden returns, and how it should look like. Any deviation or additions of this might cause Vaultwarden to not work as intended.

```http
content-security-policy: default-src 'self'; base-uri 'self'; form-action 'self'; object-src 'self' blob:; script-src 'self' 'wasm-unsafe-eval'; style-src 'self' 'unsafe-inline'; child-src 'self' https://*.duosecurity.com https://*.duofederal.com; frame-src 'self' https://*.duosecurity.com https://*.duofederal.com; frame-ancestors 'self' chrome-extension://nngceckbapebfimnlniiiahkandclblb chrome-extension://jbkfoedolllekgbhcbcoahefnbanhhlh moz-extension://* ; img-src 'self' data: https://haveibeenpwned.com ; connect-src 'self' https://api.pwnedpasswords.com https://api.2fa.directory https://app.simplelogin.io/api/ https://app.addy.io/api/ https://api.fastmail.com/ https://api.forwardemail.net ;
permissions-policy: accelerometer=(), ambient-light-sensor=(), autoplay=(), battery=(), camera=(), display-capture=(), document-domain=(), encrypted-media=(), execution-while-not-rendered=(), execution-while-out-of-viewport=(), fullscreen=(), geolocation=(), gyroscope=(), keyboard-map=(), magnetometer=(), microphone=(), midi=(), payment=(), picture-in-picture=(), screen-wake-lock=(), sync-xhr=(), usb=(), web-share=(), xr-spatial-tracking=()
referrer-policy: same-origin
x-content-type-options: nosniff
x-frame-options: SAMEORIGIN
x-xss-protection: 0
```

**Reply by BlackDex** [reply]
Also, you still not have mentioned which client on macOs you are using? are you talking about Safari, or any other browser?

**Reply by EyyupA** [reply]
Oh you mean so currently I use Safari. But I tried with Chrome also.

> I think your reverse proxy is modifying the headers which set some security features of the browsers.
Vaultwarden already sends the correct headers to ensure the correct workings of Vaultwarden via the web-vault and other clients which might need specific headers or sometimes even omitting some headers.

But why it work with the other accounts? 

Currently I get this error also:
`[Error] Unhandled error in angular
Error: Uncaught (in promise): Error: You cannot build a user key without a valid UserId, building for key biometricSettings_clientKeyHalf buildKey@https://**************//app/main.7c1287a291f3bb48f2e0.js:166:262337 dd@https://**************/app/main.7c1287a291f3bb48f2e0.js:11:244828 get@https://**************//app/main.7c1287a291f3bb48f2e0.js:11:511972 @https://**************//app/main.7c1287a291f3bb48f2e0.js:166:148441 @https://**************//app/main.7c1287a291f3bb48f2e0.js:166:145856 t@https://**************//app/polyfills.575bcaf3121104d1c55d.js:1:19106 f@https://**************//app/main.7c1287a291f3bb48f2e0.js:166:145616 @https://**************//app/main.7c1287a291f3bb48f2e0.js:72:357670 s@https://**************//app/main.7c1287a291f3bb48f2e0.js:72:352555 onInvoke@https://**************//app/vendor.0d3c724de0f2ab138712.js:2:2158933 @https://**************//app/polyfills.575bcaf3121104d1c55d.js:1:2601 @https://**************//app/polyfills.575bcaf3121104d1c55d.js:1:18807 onInvokeTask@https://**************//app/vendor.0d3c724de0f2ab138712.js:2:2158753 @https://**************//app/polyfills.575bcaf3121104d1c55d.js:1:3268 w@https://**************//app/polyfills.575bcaf3121104d1c55d.js:1:10646
P — zone.umd.js:1250
(anonymous function) — zone.umd.js:1157
(anonymous function) — zone.umd.js:1173
a — main.7c1287a291f3bb48f2e0.js:72:352618
onInvoke — ng_zone.ts:443
(anonymous function) — zone.umd.js:165
(anonymous function) — zone.umd.js:1314
onInvokeTask — ng_zone.ts:428
(anonymous function) — zone.umd.js:210
w — zone.umd.js:634
	write (main.7c1287a291f3bb48f2e0.js:166:250338)
	error (main.7c1287a291f3bb48f2e0.js:166:250128)
	handleError (main.7c1287a291f3bb48f2e0.js:11:350269)
	next (vendor.0d3c724de0f2ab138712.js:2:2250601)
	(anonymous function) (vendor.0d3c724de0f2ab138712.js:2:703161)
	(anonymous function) (vendor.0d3c724de0f2ab138712.js:2:702765)
	(anonymous function) (vendor.0d3c724de0f2ab138712.js:2:702383)
	(anonymous function) (vendor.0d3c724de0f2ab138712.js:2:699726)
	r (vendor.0d3c724de0f2ab138712.js:2:737047)
	(anonymous function) (vendor.0d3c724de0f2ab138712.js:2:699518)
	emit (vendor.0d3c724de0f2ab138712.js:2:2156520)
	(anonymous function) (polyfills.575bcaf3121104d1c55d.js:1:2602)
	onHandleError (vendor.0d3c724de0f2ab138712.js:2:2159245)
	(anonymous function) (polyfills.575bcaf3121104d1c55d.js:1:2820)
	t (polyfills.575bcaf3121104d1c55d.js:1:16779)
	(anonymous function) (polyfills.575bcaf3121104d1c55d.js:1:16867)
	w (polyfills.575bcaf3121104d1c55d.js:1:10714)`

**Reply by BlackDex** [reply]
If it works on other users accounts on the exact same hardware, then it must be something inside the vault data. Some cipher is causing an issue for some reason.

And as i'm no Typescript developer and do not really know the web-vault code that well, i can't say what is going on here.
It might be a bug fixed in a newer version of the web-vault. It might be because  you used a KDF setting which is causing issues, so, you use `Argon2id` for example, and something prevents loading that in the browser. That your vault is empty, but the org vault is visible could explain that maybe?

I would suggest to figure out since when this is happening, look for ciphers/vault items added around that period of time and try to see which vault item it is, and as a reference copy the raw data from the database as a backup and maybe for troubleshooting this issue.

And since you are able to see all items in other clients, try to open, edit and save that item via the other client and see if you vault loads again.

**Reply by BlackDex** [reply]
So, what I'm also saying is, it could still be some form of headers which restrict the browser from loading some functionality maybe. I would suggest to double check the headers and match it with the above output.

Else, check the ciphers/items in your vault, as there must be something there.

**Reply by EyyupA** [reply]
The header are same, checked with copy paste.

**Reply by EyyupA** [reply]
Doesn't VW have a debug mode? So that I can see which item is causing problems? Because checking 822 items manually is a lot.

**Reply by BlackDex** [reply]
Vaultwarden does. But the web-vault (not developed or maintained by this project) does not as far as i know. To be fair, i have not checked if there is a way to have it output more debug data at all.

**Reply by EyyupA** [reply, marked as answer]
After deleting all items from the recycle bin, I noticed that the admin panel also contains the items from the recycle bin, which means I have exported them again and imported them into the new account. I noticed that the number of items in the admin panel is now correct.
All passwords and items load correctly in the browser. 

Means it somehow looks at the account encryption why it could not load on the web.

**Reply by BlackDex** [reply]
Very strange. I have really no clue on why that would be.
Glad it is solved now (If i'm correctly reading this).
